libs/template/file.go

```diff
@@ -0,0 +1,96 @@
+package template
+
+import (
+	"context"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/databricks/cli/libs/filer"
+)
+
+// Interface for an in memory representation of a file
+type file interface {
+	// Full path of the file, in the os native format. For example /foo/bar on
+	// Unix and C:\foo\bar on windows
+	Path() string
+
+	// Unix like file path relative to the "root" of the instantiated project. Is used to
+	// evaluate whether the file should be skipped by comparing it to a list of
+	// skip glob patterns
+	RelPath() string
+
+	// This function writes this file onto the disk
+	PersistToDisk() error
+}
+
+type fileCommon struct {
```

Comment:
The attributes below all relate to the destination path. Something like `fileDestination` or `destinationPath`.

Anything with `common` in the name should be scrutinized ;-)

Reply:
done